framework/db/mssql/QueryBuilder.php

```diff
@@ -250,13 +250,48 @@ public function checkIntegrity($check = true, $schema = '', $table = '')
         return $command;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    protected function mountCommonSqlForCommentAddition($comment, $table, $column = null)
+    {
+        $tableSchema = $this->db->schema->getTableSchema($table);
+
+        $schemaName = $tableSchema->schemaName ? "N'" . $tableSchema->schemaName . "'": 'SCHEMA_NAME()';
+        $tableName = "N'{$tableSchema->name}'";
```

Comment:
Should we use `quoteTableName` here?

Reply:
We shouldn't. `fn_listextendedproperty`, `sp_addextendedproperty` and `sp_updateextendedproperty` does not accept quoted names, even althoug this is unbeliveable. I tried it in many different ways/combinations of quotations right in mssql and each trying results in error. This is one of the reasons the code was not working before.

Reply:
That's a bit weird. What would happen if you'll use `'` in a name?

Reply:
I think this happens because those "methods" are, in fact, mssql functions. and as such, they have limitations in what will be received by its params. Function params are only datatypes (varchar, int, etc). But just in case, I will take a bit more time to show you some screenshots and search any workarounds for it on the web. I think this have some impact for mssql users and so it is really necessary to take double care.

Reply:
@samdark, not even `getTableSchema()` works when using a table having a quote in its name:
![2019-06-28_00-36](https://user-images.githubusercontent.com/412490/60315434-ee808b80-993c-11e9-8391-410bb0d5df07.png)

As you can see in that debug screenshot, the table exists.

We should fix `mssql\Schema::findColumns()` line 394 `$whereSql .= " AND [t1].[table_schema] = '{$table->schemaName}'";` to quote table name.

But quotation really can not be done for `sp_listextendedproperty` or for its alternative, a query in sys.* objects, because both need a varchar data. If we use `[tbl' testes]` or `'[tbl'' testes]'` instead of `tbl'' testes` we get errors:
![2019-06-28_02-16](https://user-images.githubusercontent.com/412490/60319126-cd736700-994b-11e9-9b5a-e5b26d524b92.png)
![2019-06-28_02-20](https://user-images.githubusercontent.com/412490/60319128-ce0bfd80-994b-11e9-885f-88c84cff9aff.png)

Using just escaped name works fine:
![2019-06-28_02-22](https://user-images.githubusercontent.com/412490/60319154-e4b25480-994b-11e9-90b4-e303d288d444.png)

Reply:
For now, I only can see two options:

1. (Infeasible for Yii 2) Change the design of db objects' interations to make `createCommand()` use `bindParams()`, so that the own PDO would escape those names when necessary; OR

2. Just escape single quotes using `str_replace("'", "''", "tbl' testes")` or something like because mssql use two single quotes to generate one escaped single quote.

I think it would be valid for the most of the situations, but I feel it would break for some complex table names (or columns). Since right now the code breaks for *all* those situations, I can see this not perfect solution as a very enough improvement. 

Let me know what you think.

Reply:
Going second way is reasonable.

Reply:
AS far as I know it works with quoted table names see tests from this PR https://github.com/yiisoft/yii2/pull/17413/files
@sdlins It appears that I fixed the same issues as you. Could you review my PR and leave you comments as we need to merge them somehow either accept your changes or mine.